File: helloscala-lagom-client/src/main/scala/com/example/client/HelloLagomService.scala
package com.example.client

import com.lightbend.lagom.scaladsl.api.{ Service, ServiceCall, ServiceCallTransport }
import io.grpc.examples.helloworld.{ HelloReply, HelloRequest }


trait HelloLagomService extends Service {

  import Service._

  override def descriptor() = {
    named("helloworld.Greeter").withCalls(
      namedCall("SayHello", sayHello _)
    ).withServiceCallTransports(ServiceCallTransport.Grpc)
  }

  def sayHello(): ServiceCall[HelloRequest, HelloReply]

}

File: hellojava-lagom-client/src/main/java/com/example/client/HelloLagomService.java
package com.example.client;

import static com.lightbend.lagom.javadsl.api.Service.*;

import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import io.grpc.examples.HelloReply;
import io.grpc.examples.HelloRequest;


public interface HelloLagomService extends Service {

    ServiceCall<HelloRequest, HelloReply> sayHello();

    default Descriptor descriptor() {
        return named("helloworld.Greeter.java").withCalls(
                namedCall("SayHello", this::sayHello)
        )
                // TODO: support ServiceCallTransport.Grpc in javadsl
                //.withServiceCallTransports(ServiceCallTransport.Grpc)
                ;
    }
}